server/src/test/java/io/crate/expression/scalar/formatting/ToCharFunctionTest.java

```diff
@@ -62,6 +62,11 @@ public void testEvaluateInterval() {
         assertEvaluate("to_char(INTERVAL '1 year 2 months 3 weeks 5 hours 6 minutes 7 seconds', 'YYYY MM DD HH12:MI:SS')", "0001 03 22 05:06:07");
     }
 
+    @Test
+    public void testEvaluateIntervalMilliseconds() {
+        assertEvaluate("to_char(INTERVAL '1 year 2 months 3 weeks 5 hours 6 minutes 7 seconds 1000 milliseconds', 'YYYY MM DD HH12:MI:SS')", "0001 03 22 05:06:08");
```

Comment:
Please add a test (can be another `assertEvaluate` call in the same test) with fractional milliseconds.

Here we add 7 seconds and 1000 milliseconds and get 8. 
Would be interesting to test smth like `... 7 seconds 500 milliseconds`

Reply:
Done. Added several. 
It turns out that in `io.crate.interval.PGIntervalParser#apply(java.lang.String)` line 177 it is necessary to use `parseInteger()`, otherwise the int portion of the seconds (the 1 in 1.7) would be cut off.